### PR TITLE
Add support for `ADD_PCI_HOLES` to stage0

### PR DESCRIPTION
**Pull request by andrisaar**
QEMU itself doesn't need this, but to support some other VMMs will have broken ACPI tables if we don't patch them with this custom command.

### CODE REVIEW

File: stage0/src/acpi.rs [hunk 1]
```diff
@@ -373,12 +494,15 @@ impl RomfileCommand {
...
+            Some(CommandTag::AddPciHoles) => {
+                Ok(Command::AddPciHoles(unsafe { &self.body.pci_holes }))
+            }
             _ => Err("Invalid command tag in table-loader"),
         }
     }
 
     fn invoke(&self, fwcfg: &mut FwCfg, acpi_digest: &mut Sha256) -> Result<(), &'static str> {
-        if self.tag > 0x80000000 {
+        if self.tag > 0x80000001 {
```
Comment (tiziano88) [inline stage0/src/acpi.rs:505]: Can you refer to the constant above?

**Review by tiziano88 — APPROVED** [review]

File: stage0/src/acpi.rs (hunk 1 above)
Comment (andrisaar) [reply stage0/src/acpi.rs:505]: Is there a neat way to request the max value from an enum?

Comment (tiziano88) [reply stage0/src/acpi.rs:505]: Not that I know of, but you can define a `const MAX_VALUE = AddPciHoles` at the top and use that, to make things even clearer.

**Review by tiziano88 — APPROVED** [review]

**Review by conradgrobler — APPROVED** [review]

File: stage0/src/acpi.rs (hunk 1 above)
Comment (andrisaar) [reply stage0/src/acpi.rs]: I've changed it a bit more; the idea was that if we encounter a unknown command in the lower 31 bits, we error out and panic, whereas when we encounter an unknown vmm-specific command (in the higher 31 bits) we just log an error and continue.

That logic should be a bit more explicit now.

**Pull request merged**